dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -536,6 +539,33 @@ private static List<String> getDistinctEngineVersions(final Set<Version> workflo
             .toList();
     }
 
+    /**
+     * Returns a set of Author containing non-ORCID authors and ORCID authors with additional information.
+     * @param entry
+     * @return
+     */
+    private static Set<Author> getAllAuthors(Entry entry) {
+        Set<Author> allAuthors = new HashSet<>();
+        if (!entry.getOrcidAuthors().isEmpty()) {
+            Optional<String> token = ORCIDHelper.getOrcidAccessToken();
+            if (token.isPresent()) {
+                Set<OrcidAuthorInformation> orcidAuthorInformation = ((Set<OrcidAuthor>)entry.getOrcidAuthors()).stream()
+                        .map(orcidAuthor -> ORCIDHelper.getOrcidAuthorInformation(orcidAuthor.getOrcid(), token.get()))
```

Comment:
When we bulk index, could this cause any problems (rate limiting, unacceptable run times, etc)?  Currently, we don't have many `OrcidAuthors`, so probably not at the moment...

Reply:
The rate limit is 24 requests per second ([source](https://info.orcid.org/ufaqs/what-are-the-api-limits/)). We don't have that many ORCID authors (my db dump from April 10th says there's 38 ORCID authors), but the number of requests to ORCID would be higher than 38 because multiple entries can reference the same ORCID author, and we would make a request to ORCID for each entry during an index update.

Perhaps it's time to implement https://github.com/dockstore/dockstore/issues/4786 so there's an ORCID author info cache, which would reduce the number of calls to ORCID to the number of ORCID authors we have (example: 38).

As for run times, indexing locally without getting ORCID author info takes 1m 55s and indexing with getting ORCID author info takes 2 m 18 s, so it's not too bad.

An alternative if we don't want the overhead of getting the ORCID author name is to just not do it and only index the ORCID ID. However, this would mean that the authors facet on the search page would have ORCID IDs instead of the name.

Reply:
> Perhaps it's time to implement #4786 so there's an ORCID author info cache, which would reduce the number of calls to ORCID to the number of ORCID authors we have (example: 38).

Putting this into 1.15 for discussion

Reply:
> Perhaps it's time to implement #4786 so there's an ORCID author info cache, which would reduce the number of calls to ORCID to the number of ORCID authors we have (example: 38).

FYI, I've been debating about doing something similar for checkurl in my current ticket. Even though we have lambdas doing the pinging, I'm worried it could get flagged if we have the lambdas doing a flood of outgoing requests.